src/analysis/retail/rogue/outlaw/modules/talents/BladeRush.tsx

```diff
@@ -15,44 +33,143 @@ class BladeRush extends Analyzer {
+      this.onBladeRushUsable,
     );
+    
+    this.unusableUptimes.push({
+      start: this.owner.fight.start_time,
+      end: -1,
+    });
+  }
+
+  endUsablePeriod(timestamp: number) {
+    if(this.isInUsablePeriod)
+    {
+      this.isInUsablePeriod = false;
+      this.unusableUptimes.push({
+        start: timestamp,
+        end: -1,
+      });
+    }
   }
 
-  onFinishMove(event: CastEvent) {
-    const cpCost = getResource(event.classResources, RESOURCE_TYPES.COMBO_POINTS.id)?.cost;
-    if (!cpCost) {
+  startUsablePeriod(timestamp: number) {
+    if(!this.isInUsablePeriod)
+    {
+      this.isInUsablePeriod = true;
+      this.unusableUptimes.at(-1)!.end = timestamp;
+    }
+  }
+
+  // Record the accuracy of every Blade Rush casts
+  protected onBladeRush(event: CastEvent) {
+    let value = QualitativePerformance.Good;
+    const energy = getResource(event.classResources, RESOURCE_TYPES.ENERGY.id);
+    const tooltip = (
+      <>
+        At {this.owner.formatTimestamp(event.timestamp)} you cast 
+        <SpellLink id={TALENTS.BLADE_RUSH_TALENT}/> at {energy!.amount} energy
+      </>
+    );
+
+    if (energy!.amount > MAX_ENERGY_TRESHOLD) {
+      value = QualitativePerformance.Fail;
+    }
+    else if(energy!.amount > ACCEPTABLE_ENERGY_TRESHOLD){
+      value = QualitativePerformance.Ok;
+    }
+    this.castEntries.push({ value, tooltip });
+  }
+
+  // Record periods spent under MIN_ENERGY_TRESHOLD with BR off cd
+  protected onResourceChange(event: ResourceChangeEvent) {
+    const energy = getResource(event.classResources, RESOURCE_TYPES.ENERGY.id);
+    const energyAmount = energy?.amount;
+    if (!energy ||
+       !energyAmount ||
+       this.spellUsable.isOnCooldown(TALENTS.BLADE_RUSH_TALENT.id)) {
       return;
     }
-    if (this.spellUsable.isOnCooldown(TALENTS.BLADE_RUSH_TALENT.id)) {
-      const cooldownRemaining = this.spellUsable.cooldownRemaining(TALENTS.BLADE_RUSH_TALENT.id);
-      const extraCDR = this.selectedCombatant.hasBuff(SPELLS.TRUE_BEARING.id) ? cpCost * 1000 : 0;
-      const cooldownReduction = cpCost * 1000 + extraCDR;
-      const newChargeCDR = cooldownRemaining - cooldownReduction;
-      if (newChargeCDR < 0) {
-        this.spellUsable.endCooldown(TALENTS.BLADE_RUSH_TALENT.id);
-      } else {
-        this.spellUsable.reduceCooldown(
-          TALENTS.BLADE_RUSH_TALENT.id,
-          cooldownReduction,
-          event.timestamp,
-        );
-      }
+
+    if (energyAmount <= MIN_ENERGY_TRESHOLD) {
+      this.startUsablePeriod(event.timestamp);
+      
+    } else {
+      this.endUsablePeriod(event.timestamp);
     }
   }
+
+  // Count blade rush time spent off cd
+  onBladeRushUsable(event: UpdateSpellUsableEvent) {
+    if(event.updateType === UpdateSpellUsableType.BeginCooldown) {
+        this.endUsablePeriod(event.timestamp);
+      }
+  }
+
+  get guide(): JSX.Element {
+    this.unusableUptimes.at(-1)!.end = this.owner.fight.end_time;
```

Comment:
I would recommend not using `!` here, it will cause an error if there aren't any values in `this.unusableUptimes`. It's also manipulating state in a function that's intended to be returning a value for display. I'd suggest adding an event listener for `fightend` and doing this there:

```typescript
  constructor(options: Options) {
    // other stuff here
    this.addEventListener(Events.fightend, this.finalize);
  }

  // other stuff here

  private finalize() {
    const uptime = this.unusableUptimes.at(-1);
    if (!uptime) {
      return;
    }

    uptime.end = this.owner.fight.end_time;
  }
```